AmplifyPlugins/Auth/AWSCognitoAuthPlugin/Support/Utils/AWSCognitoAuthPluginUserDefaults.swift

```diff
@@ -0,0 +1,21 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+struct AWSCognitoAuthPluginUserDefaults {
+
+    private static let preferPrivateSessionKey = "privateSessionKey"
+
+    static func storePreferredBrowserSession(privateSessionPrefered: Bool) {
+        UserDefaults.standard.setValue(privateSessionPrefered, forKey: preferPrivateSessionKey)
```

Comment:
Are we using UserDefaults anywhere else? It would be good to abstract this into a behavior so we could inject our own user defaults provider for testing

Reply:
We are not using anywhere else in Auth currently. Moved this to a behavior.